dataplane/internal/engine/table.go

```diff
@@ -443,3 +450,45 @@ func RemoveNeighbor(ctx context.Context, c fwdpb.ServiceClient, ip []byte) error
 
 	return nil
 }
+
+// UpdatePortSrcMAC updates a port's source mac address.
+func UpdatePortSrcMAC(ctx context.Context, c fwdpb.ServiceClient, portName string, mac []byte) error {
```

Comment:
(no action required on this PR): would be nice if we eventually have a diagram of how each of these tables work together for packet processing.